lms/djangoapps/courseware/views/views.py

```diff
@@ -374,12 +374,11 @@ def get_last_accessed_courseware(course, request, user):
         if SelfPacedConfiguration.current().enable_course_home_improvements:
             context['last_accessed_courseware_url'] = get_last_accessed_courseware(course, request, user)
 
-        now = datetime.now(UTC())
-        effective_start = _adjust_start_date_for_beta_testers(user, course, course_key)
-        if not in_preview_mode() and staff_access and now < effective_start:
+        if is_preview_menu_disabled(user, course):
             # Disable student view button if user is staff and
             # course is not yet visible to students.
             context['disable_student_access'] = True
+            context['supporst_preview_menu'] = False
```

Comment:
Typo: `supporst_preview_menu`
Also, I asked if `supports_preview_menu` could be renamed as `preview_menu_enabled`.

Also, seems to point to missing unit tests, where you could mock is_preview_menu_disabled.